File: example/javalib/testing/4-test-grouping/foo/testng/src/foo/WorldTests.java
package foo;

import static org.testng.Assert.assertTrue;
import org.testng.annotations.Test;

public class WorldTests {
  @Test
  public void world() throws Exception {
    System.out.println("Testing World");
    String result = new Foo().hello();
    assertTrue(result.endsWith("World"));
    Thread.sleep(1000);
    System.out.println("Testing World Completed");
  }
}

File: example/scalalib/testing/5-test-grouping/foo/test/src/WorldTests.scala
package foo
import utest.*
object WorldTests extends TestSuite {
  def tests = Tests {
    test("world") {
      println("Testing World")
      val result = Foo.hello()
      assert(result.endsWith("World"))
      Thread.sleep(1000)
      println("Testing World Completed")
      result
    }
  }
}
